src/Moq/Interception/InterceptionAspects.cs

```diff
@@ -186,63 +187,25 @@ public static bool Handle(Invocation invocation, Mock mock)
 			}
 			
 			MethodInfo invocationMethod = invocation.Method;
-			if (invocationMethod.IsPropertyAccessor())
+			if (!invocationMethod.IsPropertyAccessor())
 			{
-				string propertyNameToSearch = invocationMethod.Name.Substring(AccessorPrefixLength);
-				PropertyInfo property = invocationMethod.DeclaringType.GetProperty(propertyNameToSearch, Type.EmptyTypes);
-
-				if (property == null)
-				{
-					return false;
-				}
-
-				var expression = GetPropertyExpression(invocationMethod.DeclaringType, property);
+				return false;
+			}
 
-				object propertyValue;
+			string propertyName = invocationMethod.Name.Substring(AccessorPrefixLength);
+			PropertyInfo property = invocationMethod.DeclaringType.GetProperty(propertyName, Type.EmptyTypes);
+			Debug.Assert(property != null);
 
-				Setup getterSetup = null;
-				if (property.CanRead(out var getter))
-				{
-					if (ProxyFactory.Instance.IsMethodVisible(getter, out _))
-					{
-						propertyValue = CreateInitialPropertyValue(mock, getter);
-						getterSetup = new StubbedPropertyGetterSetup(mock, expression, getter, () => propertyValue);
-						mock.MutableSetups.Add(getterSetup);
-					}
```

Comment:
Those visibility checks are no longer required because we already know that either the getter or the setter can be intercepted by the proxy factory; otherwise we wouldn't end up here in the first place.

If the other accessor isn't interceptable, that amounts to the same thing as if it were completely absent: we'll still create a setup just for the present accessor (even if it won't be of much actual use beyond preventing this code from executing repeatedly for the same property).